README.md

```diff
@@ -29,95 +33,319 @@ are well established for RSE-ops, and which require further development. We can
+> to. You have to know how to administer a linux system -- or at least a container. So a lot more
+> freedom but in many of our users' eyes a lot more complexity too. That's changing -- they're
+> getting used to containers. But VMs, AWS stuff is still a maze to many of them.
+
 
 ### What about testing?
 
 While most centers do some kind of kernel and application benchmarking [11], there is a huge divide between how the scientific user tests their own code or software and using it on the shared resource. The reproducibility crisis that grew in the early 2010s [12] did advocate for more use of version control [13,14] for research software collaboration and development, and the ease of integration of continuous integration (CI) services for automated testing and deployment [15,16,17] made it more common for researchers to test and deploy their scientific software, often to package managers (e.g., pip) or containers that could be pulled and used on a cluster. However, this test and deployment process was notably separate from the HPC resources. It simply was not possible to test codes on all of the possible semi-customized HPC environments on which they might be run. Understandably there would be security issues connecting an external service to a shared resources, however this does not fully excuse the lack of innovation in this area. It should be easier for a researcher to test and even deploy their codes on or beside the resource that they ultimately will be used.
 
+> Yeah agree with this. Cloud resources/CI runners are really needed. Centers also have a hard time
+> making arguments to justify cycles on test clusters (though we do have dedicated ones
+> internally). Supporting teams who need to test on LC-relevant machines for projects that live on
+> GitHub was an ISCP project of Greg's this year -- we built a cluster in the green zone to support
+> it.  We should talk about this more -- this was an ISCP project that I proposed.
+
+
 ### What about scaling?
 
 High performance computing was designed for scaling, but it must be done intentionally. By way of job managers, MPI programming, and
-extreme parallelism [^8], it's relatively trivial to scale a task on an HPC cluster. However, the user has to know what they are doing. A related issue is testing at scale. As tests need to be run frequently and across many different architectures, the incentive structure is arguably not there to maintain such a testing framework. This is in contrast to some Cloud Native tools that handle scaling or scaled testing for the user. This suggests that the greatest challenges are not related to the technologies, but educating the user base and RSEs how to properly use them. There is a huge opportunity for development of RSE-ops tools that can better automate this scaling.
+extreme parallelism [^8], it's relatively trivial to scale a task on an HPC cluster. However, the user has to know what they are doing.
+
+> Don't these two sentences contradict each other?
```

Comment:
I don't think so - I'm saying scaling is native to HPC, but you have to know how to properly use the tools to do it. A new person shelling into a login node is going to struggle a bit.

Reply:
yeah that sounds good.